ush/om_data.py

```diff
@@ -0,0 +1,390 @@
+                                       '.tar.gz')
+                go_test = os.path.join(self.data_src, 'gdas.' + self.pdate[:-2], self.pdate[-2:],
+                                       'atmos', mon, mon + '_' + self.data_subtype + '.tar.gz')
+                lm_test = os.path.join(self.data_src, 'gdas.' + self.pdate[:-2], self.pdate[-2:],
+                                       mon, mon + '_' + self.data_subtype + '.tar.gz')
+
+            case 'minmon':
+                gw_test = os.path.join(self.data_src, 'gdas.' + self.pdate[:-2], self.pdate[-2:],
+                                       'products', 'atmos', mon)
+                go_test = os.path.join(self.data_src, 'gdas.' + self.pdate[:-2], self.pdate[-2:],
+                                       'atmos', mon)
+                lm_test = os.path.join(self.data_src, 'gdas.' + self.pdate[:-2], self.pdate[-2:],
+                                       mon)
+
+            # conmon is only in obs-mon and leg-mon structures
+            case 'conmon':
+                lm_test = os.path.join(self.data_src, 'gdas.' + self.pdate[:-2], self.pdate[-2:],
+                                       mon, self.data_subtype, self.ctl_file[0])
+
+        if os.path.exists(om_test):
+            self.dir_struct = 'obs-mon'
+        elif os.path.exists(gw_test):
+            self.dir_struct = 'glb-wkflw'
+        elif os.path.exists(go_test):
+            self.dir_struct = 'gfs-ops'
+        elif os.path.exists(lm_test):
+            self.dir_struct = 'leg-mon'
+
+# --------------------------------------------------------------------------------------------
+
+    def get_rad_datafile(self, cycle, file, dest_dir):
+
+        """
+        Locate and copy file to dest_dir.
+
+        Legacy radmon data is unique in that it is stored in tar files (to reduce file
+        count).  There are 4 files for each cycle: radmon_angle.tar.gz, radon_bcoef.tar.gz,
+        radmon_bcor.tar.gz, and radmon_time.tar.gz.  The input file name will be extracted
+        (if present) from the specified tar file.
+
+        Note that data in obs-mon structure (used only for development) is simply a
+        directory containing uncompressed files (not stored in a tar file).
+
+        Parameters:
+            cycle (str):   cycle time (in YYYYMMDDHH format) of the requested tar file
+            file (str):    data file name
+            dest_dir (path):  full directory path to where data file is to be extracted.
+        Return:
+            None
+        """
+
+        pdy = cycle[:-2]
+        hh = cycle[-2:]
+
+        match self.dir_struct:
+            case 'obs-mon':
+                tarfile = ""
+                f = os.path.join(self.data_src, 'rad_data', file)
+                if os.path.isfile(f):
+                    shutil.copy(f, dest_dir)
+
+            case 'glb-wkflw':
+                tarfile = os.path.join(self.data_src, 'gdas.' + pdy, hh, 'products', 'atmos',
```

Comment:
this is closer to what I was expecting, but would probably go further and define a path that can then be further 'joined'

Reply:
Ok, that makes sense.  I'll make it so.